src/object/obj_tx.c

```diff
@@ -3214,17 +3215,34 @@ dc_tx_convert(struct dc_object *obj, enum obj_rpc_opc opc, tse_task_t *task)
 	args->th = dc_tx_ptr2hdl(tx);
 	args->flags = 0;
 
+	/*
+	 * Make the @task to depend on the @tx_task before dc_tx_attach(), that will prevent
+	 * the @task being released after dc_tx_attach() returned because of the dependency.
+	 */
 	rc = dc_task_depend(task, 1, &tx_task);
 	if (rc != 0) {
 		D_ERROR("Fail to add dep on TX convert task: "DF_RC"\n",
 			DP_RC(rc));
 		goto out;
 	}
 
+	dep = true;
+
+	rc = dc_tx_attach(dc_tx_ptr2hdl(tx), obj, opc, task);
+	obj = NULL;
+	if (rc < 0) {
+		D_ERROR("Fail to attach TX for opc %u: "DF_RC"\n",
+			opc, DP_RC(rc));
+		goto out;
+	}
+
+	/* Got failure from dc_tx_attach(), maybe because of existence check. */
+	if (task->dt_result != 0)
+		D_GOTO(out, rc = task->dt_result);
```

Comment:
just confirm does it mean the existence check task(s) created inside dc_tx_attach was executed and completed by other threads and the err propogate to task, here check the result?
does the below logic depend on the check at above L3240? but that case it possibly happen at other time at below L3243 ~ L 3255, so seems unnecessary for the check here?